api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/mfa/trusteddevice/BaseDeviceFingerprintComponentProperties.java

```diff
@@ -0,0 +1,33 @@
+package org.apereo.cas.configuration.model.support.mfa.trusteddevice;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.Serializable;
+
+/**
+ * Common configuration for device fingerprint components.
+ *
+ * @author Daniel Frett
+ * @since 5.3.0
+ */
+@Getter
+@Setter
+public abstract class BaseDeviceFingerprintComponentProperties implements Serializable {
```

Comment:
Add the annotation to generate a no-args constructor. Add to all subclasses as well.